src/Application/Actions/Donations/Confirm.php

```diff
@@ -105,17 +110,24 @@ protected function action(Request $request, Response $response, array $args): Re
         \assert($paymentMethodId !== ""); // required to call updatePaymentMethodBillingDetail
 
         if (!$donation->hasExpectedMatchingReserved()) {
-            $this->entityManager->rollback();
-            $this->logger->warning(sprintf(
-                'Donation %s does not have expected match funds reserved at confirmation',
-                $donation->getUuid(),
-            ));
-            return new JsonResponse([
-                'error' => [
-                    'message' => 'Donation does not have expected match funds reserved',
-                    'code' => 'donation-no-match-funds-reserved',
-                ],
-            ], 400);
+            if ($this->enableNoReservationsMode) {
+                $this->logger->warning(sprintf(
```

Comment:
```suggestion
                $this->logger->notice(sprintf(
```
Probably helpful to be able to see if there are other warnings and not create a high volume of these when we already know we've flipped the switch.